src/App.tsx

```diff
@@ -1,51 +1,128 @@
 import React from 'react';
 import './App.scss';
+import classNames from 'classnames';
+import goodsFromServer from './goods';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
-};
+interface State {
+  selectedGoods: string[];
+  goods: string[];
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+    goods: goodsFromServer,
+  };
+
+  getHeaderTitle() {
+    const { selectedGoods } = this.state;
+
+    if (!selectedGoods.length) {
+      return 'No goods selected';
+    }
+
+    if (selectedGoods.length === 1) {
+      return `${selectedGoods[0]} is selected`;
+    }
+
+    const lastIndex = selectedGoods.length - 1;
+    const firstPart = selectedGoods.slice(0, -1);
+    const joined = `${firstPart.join(', ')} and ${selectedGoods[lastIndex]}`;
+
+    return `${joined} are selected`;
+  }
+
+  isSelected(goodName: string) {
+    const { selectedGoods } = this.state;
+
+    return selectedGoods.includes(goodName);
+  }
+
+  selectGood(good: string) {
+    const { selectedGoods } = this.state;
+
+    this.setState({ selectedGoods: [...selectedGoods, good] });
+  }
+
+  unselectGood(good: string) {
+    const { selectedGoods } = this.state;
+    const newGoods = selectedGoods.filter(el => el !== good);
+
+    this.setState({ selectedGoods: newGoods });
+  }
+
+  selectToggle(goodName: string) {
+    if (this.isSelected(goodName)) {
+      this.unselectGood(goodName);
+
+      return;
+    }
+
+    this.selectGood(goodName);
```

Comment:
Do you really need to return nothing?
```suggestion
    this.isSelected(goodName)
    	? this.unselectGood(goodName)
    	: this.selectGood(goodName);
```